sympy/physics/control/lti.py

```diff
@@ -1422,3 +1445,545 @@ def _eval_rewrite_as_TransferFunction(self, num, den, **kwargs):
+        >>> tf3 = TransferFunction(3, 4, p)
+        >>> tfm1 = TransferFunctionMatrix([[tf1, -tf2]])
+        >>> tfm1.shape
+        (1, 2)
+        >>> tfm2 = TransferFunctionMatrix([[-tf2, tf3], [tf1, -tf1]])
+        >>> tfm2.shape
+        (2, 2)
+
+        """
+        return self._expr_mat.shape
+
+    def __neg__(self):
+        neg = -self._expr_mat
+        return _to_TFM(neg, self.var)
+
+    def __getitem__(self, key):
+        trunc = self._expr_mat.__getitem__(key)
+        if isinstance(trunc, ImmutableMatrix):
+            return _to_TFM(trunc, self.var)
+        else:
+            return TransferFunction.from_rational_expression(trunc, self.var)
+
+    def transpose(self):
+        transposed_mat = self._expr_mat.transpose()
+        return _to_TFM(transposed_mat, self.var)
+
+    def doit(self):
+        """
+        Returns the equivalent ``TransferFunctionMatrix`` with each ``Series`` and ``Parallel``
+        element rewritten as ``TransferFunction``.
+
+        Examples
+        ========
+
+        >>> from sympy.abc import s
+        >>> from sympy.physics.control.lti import TransferFunction, TransferFunctionMatrix, Series, Parallel
+        >>> from sympy.printing import pprint
+        >>> tf_1 = TransferFunction(1, s, s)
+        >>> tf_2 = TransferFunction(1, s**2, s)
+        >>> tfm_1 = TransferFunctionMatrix([[Series(tf_1, tf_2), tf_1], [tf_2, Parallel(tf_1, tf_2)]])
+        >>> tfm_1
+        TransferFunctionMatrix(((Series(TransferFunction(1, s, s), TransferFunction(1, s**2, s)), TransferFunction(1, s, s)), (TransferFunction(1, s**2, s), Parallel(TransferFunction(1, s, s), TransferFunction(1, s**2, s)))))
+        >>> pprint(tfm_1, use_unicode=False)
+        [ 1      1   ]
+        [----    -   ]
+        [   2    s   ]
+        [s*s         ]
+        [            ]
+        [ 1    1    1]
+        [ --   -- + -]
+        [  2    2   s]
+        [ s    s     ]
+        >>> tfm_1.doit()
+        TransferFunctionMatrix(((TransferFunction(1, s**3, s), TransferFunction(1, s, s)), (TransferFunction(1, s**2, s), TransferFunction(s**2 + s, s**3, s))))
+        >>> pprint(_, use_unicode=False)
+        [1     1   ]
+        [--    -   ]
+        [ 3    s   ]
+        [s         ]
+        [          ]
+        [     2    ]
+        [1   s  + s]
+        [--  ------]
+        [ 2     3  ]
+        [s     s   ]
+
+        """
+        doit_arg = []
+        for i in self.args[0]:
+            temp = []
+            for j in i:
+                if not isinstance(j, TransferFunction):
+                    j = j.doit()
+                temp.append(j)
+            doit_arg.append(temp)
+            temp = []
+        return TransferFunctionMatrix(doit_arg)
+
+    def elem_poles(self):
+        """
+        Returns the poles of each element of the ``TransferFunctionMatrix``.
+
+        .. note::
+            Actual poles of a MIMO system are NOT the poles of individual elements.
```

Comment:
I don't understand this. If you say the following is true:
>  Actual poles of a MIMO system are NOT the poles of individual elements.

Then why are you doing it this way? I don't think we need two methods to find poles.

Reply:
Firstly, what I meant was - _Actual poles of a MIMO system are NOT **always** the poles of individual elements_. For an uncoupled TFM, the poles of the system are **poles of the individual components**. It might be beneficial for the user to visualize all the poles and zeros of the elements at once. Later, when adding the Poles and Zeros of the system, a beginner user might find it convenient to compare the poles and zeros to element-wise poles and zeros and get an intuitive understanding of the MIMO system stability. We can add an example in the textbook problems section where we could ask the user to find the poles and zeros of the system and compare them with the poles and zeros of the individual elements. In my opinion, these methods have practical use cases, and we should keep them.